spec/integration/users_spec.rb

```diff
@@ -1,124 +1,20 @@
-      expect(page).to have_content('Tom')
-      expect(page).to have_content('Lilly')
-    end
-
-    it 'should display the number of posts for each user' do
-      visit root_path
-      expect(page).to have_content('Number of posts: 2')
-      expect(page).to have_content('Number of posts: 2')
-    end
-
-    it 'should display profile pic for each user' do
-      visit root_path
-      users = User.all.order(:id)
-      profile_pic = page.all('img')
-      expect(profile_pic[0][:src]).not_to be('')
-      expect(profile_pic.length).to eq(users.length)
-    end
-  end
-
-  it 'When  clicks on a user, it should redirected to that user show page' do
+RSpec.describe 'user_index', type: :feature do
+  before(:each) do
+    @user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')
     visit root_path
-    click_link 'Tom'
-    expect(current_path).to eq(user_path(User.first))
-  end
-
-  # user show page:
-
-  context 'User show page' do
-    before(:each) do
-      visit user_path(@first_user.id)
-    end
-    it 'should show the user profile picture' do
-      visit user_path(@first_user.id)
-      users = User.all.order(:id)
-      profile_pic = page.all('img')
-      expect(profile_pic[0][:src]).not_to be('')
-      expect(profile_pic.length).to eq(users.length - 1)
-    end
-
-    it 'should display user/s username ' do
-      visit user_path(@first_user.id)
-      expect(page).to have_content('Tom')
-    end
-  end
-
-  it 'should display the number of posts the user has written' do
-    visit user_path(@first_user.id)
-    expect(page).to have_content('Number of posts: 2')
-  end
-
-  it 'should display users bio' do
-    visit user_path(@first_user.id)
-    expect(page).to have_content('Teacher from Mexico.')
-  end
-
-  it 'should display the user/s first 3 posts.' do
-    visit user_path(@first_user)
-    expect(page).to have_content('Hello')
-    expect(page).to have_content('software developer')
-    expect(page).to have_content('Hello')
-  end
-
-  it 'should display button that lets me view all of a user/s posts' do
-    visit user_path(@first_user)
-    expect(page).to have_content('see all post')
   end
-
-  it 'When  click a user/s post, it should redirects  to that post/s show page' do
-    visit user_path(@first_user)
-    click_link 'Hello'
-    expect(current_path).to eq(user_post_path(@first_user.id, @first_user.posts.first.id))
+  it 'shows the username of the users' do
+    expect(page).to have_content('Tom')
   end
-
-  it 'When I click to see all posts, it redirects me to the user/s post/s index page' do
-    visit user_path(@first_user)
-    click_link 'see all posts'
-    expect(current_path).to eq(user_posts_path(@first_user.id))
+  # it 'shows the photos of the users' do
+  #   expect(page.html).to include('avatal')
+  # end
+  it 'shows the number of post of each user' do
+    expect(page).to have_content("Number of Posts: #{@user.posts_counter}")
```

Comment:
- [ ] Kindly delete commented code to follow ruby practise.
- Kindly open this test. It is [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=I%20can%20see%20the%20profile%20picture%20for%20each%20user.)